### DISCUSSION TITLE
how do i check if the mode is spa mode or capacitor

### DISCUSSION THREAD

**Discussion by eimihar** [quasarframework/quasar#16293] [CLI - Capacitor / Cordova modes] [answered]
or something like platform.quasar.is.spa

I tried using process.env in one of my component but it's not available through quasar build, or even after i import (import * as process from "process")

Thanks!

**Comment by metalsadman** [comment]
how are you using it? https://quasar.dev/quasar-cli-vite/handling-process-env#introduction 
https://quasar.dev/options/platform-detection#platform-api

**Reply by eimihar** [reply]
thank you for the reply.. i am sorry, this is an error on my end, after tried multiple quasar build process.env.MODE returns consistently (on spa and capacitor build) (error cause i didn't manually run npm i process in the project, assumed that it's already included)

just need to confirm that the process.env seems to be empty when developing through the `quasar dev` command. is this intended?

Thanks!

**Reply by rstoenescu** [reply, marked as answer]
The "process.env" is not empty. For security reasons, trying to print it out as is will indeed show nothing. However, when used in its full form (`process.env.<...something...>` it will be replaced with the correct content and you will be able to print it out). More information is available in the "handling process env" page under the "Troubleshooting" section ([for q/app-vite](https://quasar.dev/quasar-cli-vite/handling-process-env#troubleshooting) or [for q/app-webpack](https://quasar.dev/quasar-cli-webpack/handling-process-env#troubleshooting).

![image](https://github.com/quasarframework/quasar/assets/4665641/dd3c7b7a-6e75-4251-a2d6-7258cdebcd45)
